kitsune/search/api.py

```diff
@@ -32,6 +32,15 @@ def suggest(request):
         errors['q'] = 'This field is required.'
     if product is not None and not Product.objects.filter(slug=product).exists():
         errors['product'] = 'Could not find product with slug "{0}".'.format(product)
+    if locale not in settings.SUMO_LANGUAGES:
+        if settings.NON_SUPPORTED_LOCALES.get(locale):
+            errors['locale'] = (
+                'Locale "{0}" is not supported, but has fallback locale "{1}".'.format(
+                    locale, settings.NON_SUPPORTED_LOCALES[locale])
+            )
```

Comment:
Oh, wow--that's completely not clear from the comment for that setting. I had no idea.

So is this correct:

```
if locale not in settings.SUMO_LANGUAGES:
    if locale in settings.NON_SUPPORTED_LOCALES:
        errors['locale'] = (
            'Locale "{0}" is not supported, but has fallback locale "{1}".'.format(
                locale, settings.NON_SUPPORTED_LOCALES.get(locale, settings.WIKI_DEFAULT_LANGUAGE)
            )
        )
    ...
```